spec/app/ics-101-interchain-swap/README.md

```diff
@@ -471,158 +503,215 @@ The function `sendInterchainIBCSwapDataPacket` must be invoked by a transaction
+      return makeMultiDepositResEncoded;
+
+    case "TAKE_MULTI_DEPOSIT":
+      const takeMultiDepositMsg: MsgTakeMultiAssetDepositRequest = protobuf.decode(
+        MsgTakeMultiAssetDepositRequest,
+        data.Data
+      );
+      const takeMultiDepositRes = k.OnTakeMultiAssetDepositReceived(takeMultiDepositMsg, data.stateChange);
+      const takeMultiDepositResEncoded = protobuf.encode(takeMultiDepositRes);
+      return takeMultiDepositResEncoded;
+
+    case "MULTI_WITHDRAW":
+      const multiWithdrawMsg: MsgMultiAssetWithdrawRequest = protobuf.decode(MsgMultiAssetWithdrawRequest, data.Data);
+      const multiWithdrawRes = k.OnMultiAssetWithdrawReceived(multiWithdrawMsg, data.stateChange);
+      const multiWithdrawResEncoded = protobuf.encode(multiWithdrawRes);
+      return multiWithdrawResEncoded;
+
+    case "LEFT_SWAP":
+    case "RIGHT_SWAP":
+      const swapMsg: MsgSwapRequest = protobuf.decode(MsgSwapRequest, data.Data);
+      const swapRes = k.OnSwapReceived(swapMsg, data.stateChange);
+      const swapResEncoded = protobuf.encode(swapRes);
+      return swapResEncoded;
+
+    default:
+      return;
+  }
 }
 ```
 
 `onAcknowledgePacket` is called by the routing module when a packet sent by this module has been acknowledged.
 
 ```ts
-
 // OnAcknowledgementPacket implements the IBCModule interface
-function OnAcknowledgementPacket(
-    packet: channeltypes.Packet,
-    ack channeltypes.Acknowledgement,
-)  {
-
-    var ack channeltypes.Acknowledgement
-    if (!ack.success()) {
-        refund(packet)
-    } else {
-        const swapPacket = protobuf.decode(packet.data)
-        switch swapPacket.type {
-        case CreatePool:
-            onCreatePoolAcknowledged(msg)
-            break;
-        case SingleAssetDeposit:
-            onSingleAssetDepositAcknowledged(msg)
-            break;
-        case MultiAssetDeposit:
-            onMultiAssetDepositAcknowledged(msg)
-            break;
-        case SingleAssetWithdraw:
-            onSingleAssetWithdrawAcknowledged(msg)
-            break;
-        case MultiAssetWithdraw:
-            onMultiAssetWithdrawAcknowledged(msg)
-            break;
-        case Swap:
-            var msg: MsgSwapRequest = protobuf.decode(swapPacket.data)
-            if(msg.SwapType === SwapType.Left) {
-                onLeftSwapAcknowledged(msg)
-            }else{
-                onRightSwapAcknowledged(msg)
-            }
-            break;
-        }
-    }
 
-    return nil
+// OnAcknowledgementPacket processes the packet acknowledgement and performs actions based on the acknowledgement type
+function OnAcknowledgementPacket(packet: Packet, data: IBCSwapPacketData, ack: Acknowledgement) {
+  switch (ack.response.type) {
+    case "Acknowledgement_Error":
+      return store.refundPacketToken(packet, data);
+    default:
+      switch (data.type) {
+        case "MAKE_POOL":
+          const msgMakePool: MsgMakePoolRequest = protobuf.decode(data.Data);
```

Comment:
Do we need change this to json marshall to in favor of wasm implementing.

Reply:
I'm not sure, when I use json marshal, it was failed in previous time in ic-100